ui/src/shared/components/VisOptionsButton.tsx

```diff
@@ -0,0 +1,73 @@
+// Libraries
+import React, {Component} from 'react'
+import {connect} from 'react-redux'
+
+// Components
+import {Button, ButtonShape, IconFont, ComponentColor} from 'src/clockface'
+
+// Actions
+import {setActiveTab} from 'src/shared/actions/v2/timeMachines'
+
+// Utils
+import {getActiveTimeMachine} from 'src/shared/selectors/timeMachines'
+
+// Types
+import {TimeMachineTab} from 'src/types/v2/timeMachine'
+import {AppState} from 'src/types/v2'
+
+interface StateProps {
+  activeTab: TimeMachineTab
+}
+
+interface DispatchProps {
+  onSetActiveTab: typeof setActiveTab
+}
+
+interface OwnProps {}
+
+type Props = StateProps & DispatchProps & OwnProps
+
+class VisOptionsButton extends Component<Props> {
+  public render() {
+    const {activeTab} = this.props
+
+    const color =
+      activeTab === TimeMachineTab.Visualization
+        ? ComponentColor.Primary
+        : ComponentColor.Default
+
+    return (
+      <Button
+        color={color}
+        shape={ButtonShape.Square}
+        icon={IconFont.CogThick}
+        onClick={this.handleClick}
+      />
+    )
+  }
+
+  private handleClick = (): void => {
+    const {onSetActiveTab, activeTab} = this.props
+
+    if (activeTab === TimeMachineTab.Queries) {
+      onSetActiveTab(TimeMachineTab.Visualization)
+    } else {
+      onSetActiveTab(TimeMachineTab.Queries)
+    }
+  }
+}
+
+const mstp = (state: AppState) => {
+  const {activeTab} = getActiveTimeMachine(state)
```

Comment:
could you type mstp and mdtp with StateProps and OwnProps?

Reply:
Good catch